src/components/base/outline-calendar-link-dropdown/outline-calendar-link-dropdown.ts

```diff
@@ -0,0 +1,221 @@
+import { CSSResultGroup, TemplateResult, html } from 'lit';
+import { OutlineElement } from '../outline-element/outline-element';
+import '../outline-icon/outline-icon';
+import '../outline-link/outline-link';
+import { google, outlook, ics, CalendarEvent } from 'calendar-link';
+import { UnitType } from 'dayjs';
+import { customElement, property, state } from 'lit/decorators.js';
+import componentStyles from './outline-calendar-link-dropdown.css.lit';
+
+export type Duration = [number, UnitType];
+
+/**
+ * The Calendar Dropdown Link component
+ * @element outline-calendar-link-dropdown
+ */
+@customElement('outline-calendar-link-dropdown')
+export class OutlineCalendarLinkDropdown extends OutlineElement {
+  static styles: CSSResultGroup = [componentStyles];
+
+  /**
+   * The event title.
+   * Required
+   */
+  @property({ type: String })
+  title: string;
+
+  /**
+   * The event start time.
+   * Required
+   * acceptable values are JS Date | ISO 8601 string | Unix Timestamp
+   */
+  @property({ type: String })
+  start: string;
+
+  /**
+   * The event end time. You must provide either end, duration, or allDay
+   * acceptable values are JS Date | ISO 8601 string | Unix Timestamp
+   */
+  @property({ type: String })
+  end: string;
+
+  /**
+   * The event duration. You must provide either end, duration, or allDay
+   * format is an array with value(number) and unit(string) i.e [3, "hour"]
+   */
+  @property({ type: Array })
+  duration: Duration;
+
+  /**
+   * Whether this is an all day event. You must provide either end, duration, or allDay
+   */
+  @property({ type: Boolean, attribute: 'all-day' })
+  allDay: boolean;
+
+  /**
+   * Event description
+   */
+  @property({ type: String })
+  description: string;
+
+  /**
+   * Event location
+   */
+  @property({ type: String })
+  location: string;
+
+  /**
+   * Whether to mark on calendar as busy
+   */
+  @property({ type: Boolean })
+  busy: boolean;
+
+  /**
+   * Emails of other guests. An array of emails(string).
+   * i.e ['user1@test.com', 'user2@test.com']
+   */
+  @property({ type: Array })
+  guests: string[];
+
+  /**
+   * Calendar document URL
+   */
+  @property({ type: String })
+  url: string;
+
+  @state()
+  open = false;
+
+  @state()
+  event: CalendarEvent;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.event = {
+      title: this.title,
+      start: this.start,
+      end: this.end,
+      duration: this.duration,
+      allDay: this.allDay,
+      description: this.description,
+      location: this.location,
+      busy: this.busy,
+      guests: this.guests,
+      url: this.url,
+    };
+  }
+
+  render(): TemplateResult | undefined {
+    const iconName = this.open ? 'chevron-up' : 'chevron-down';
+    const menuState = this.open ? 'open' : 'hidden';
+
+    return html`
+      <div class="button-wrapper ${menuState}">
+        <button
```

Comment:
I'm a bit new to Outline and the general modus operandi here, but would it make sense to use `outline-button` here, which also includes the option to add icons? Or is it the fact that it changes behavior on opening and clicking that causes some problems? If it were possible to use that component, we'd be able to cut down on possible duplication for styles, etc.